Exercises/common.cmake

```diff
@@ -6,4 +6,18 @@ if(SPACK_CXX)
   set(ENV{CXX} ${SPACK_CXX})
 endif()
 
+include(FetchContent)
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.24)
+  # try find_package first before trying to download Kokkos
+  set(FETCH_CONTENT_EXTRA_ARGS FIND_PACKAGE_ARGS NAMES Kokkos)
+endif()
+FetchContent_Declare(
+  Kokkos
+  GIT_REPOSITORY https://github.com/kokkos/kokkos.git
+  GIT_TAG        4.0.01
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../dep/Kokkos
+  ${FETCH_CONTENT_EXTRA_ARGS}
+)
+FetchContent_MakeAvailable(Kokkos)
+
 find_package(Kokkos REQUIRED)
```

Comment:
Is this still necessary after `FetchContent_MakeAvailable`?
```suggestion
```

Reply:
It can be removed, @dalg24 asked for me to push some changes so I will once I get write access :P

Reply:
The intent with that line was that you can still build against an external Kokkos and skip the download if Kokkos.

Reply:
To skip the download (with CMake < 3.24) `find_package` has to be called first before `FetchContent_MakeAvailable`, possibly like this:
```CMake
find_package(Kokkos)
if(Kokkos_FOUND)
  message(STATUS "Using Kokkos found in ${Kokkos_DIR}")
else()
  message(STATUS "No installed Kokkos found, fetching from GitHub")
  include(FetchContent)
  (the rest of FetchContent code ...)
endif()
```